app/src/main/java/com/tsu/wordsfactory/utils/NetworkUtils.kt

```diff
@@ -0,0 +1,54 @@
+package com.tsu.wordsfactory.utils
+
+import android.app.DownloadManager
+import android.content.Context
+import android.media.AudioAttributes
+import android.media.MediaPlayer
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.net.Uri
+import android.util.Log
+import java.io.File
+import java.io.IOException
+
+class NetworkUtils(val context: Context) {
+    fun isOnline(): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val capabilities =
+            connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+        if (capabilities != null) {
+            if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
+                Log.i("Internet", "NetworkCapabilities.TRANSPORT_CELLULAR")
+                return true
+            } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
+                Log.i("Internet", "NetworkCapabilities.TRANSPORT_WIFI")
+                return true
+            } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+                Log.i("Internet", "NetworkCapabilities.TRANSPORT_ETHERNET")
+                return true
+            }
+        }
+        return false
+    }
+
+    fun playContentUri(uri: Uri) {
+        var mMediaPlayer: MediaPlayer? = null
```

Comment:
``Media Player` isn't network function at all. Yes it uses internet but this class that should be represented to themselves. Move this to other `Media` class if you need this but for real you could move this function to private fun inside `ViewHolder`